libkbfs/fs_rpc.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2016 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"fmt"
+
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol"
+	"golang.org/x/net/context"
+)
+
+type fsRPC struct {
+	config Config
+	log    logger.Logger
+}
+
+func (f fsRPC) favorites(ctx context.Context, path Path) (keybase1.ListResult, error) {
+	favs, err := f.config.KBFSOps().GetFavorites(ctx)
+	if err != nil {
+		return keybase1.ListResult{}, err
+	}
+	files := []keybase1.File{}
+	for _, fav := range favs {
+		if (fav.Public && path.Public) || (!fav.Public && !path.Public) {
+			files = append(files, keybase1.File{Name: fav.Name})
+		}
+	}
+	return keybase1.ListResult{Files: files}, nil
+}
+
+func (f fsRPC) tlf(ctx context.Context, path Path) (keybase1.ListResult, error) {
+	files := []keybase1.File{}
+
+	node, de, err := path.GetNode(ctx, f.config)
```

Comment:
Probably not for this PR, but note that eventually you're going to want to cache these `Node` objects somewhere, and possibly add a `Forget` RPC to remove them from the cache when the app is done with them.  Calling `GetNode` every time is needlessly expensive.  Although I guess it depends on the exact semantics we want for the mobile app, so we can cross that bridge later.  Just something to keep in mind.